### PR TITLE
`/report` works

### LINKED ISSUE #390: Generate timesheet report for user
**Who are you?**
CSE Dev

**What feature do you want?**
The user should be able request report on their timesheet and pomodoro  for a defined date range or a default range

**Why do you want this feature?**
To be able to view user's report over a time period.

### PR DESCRIPTION
**Pull request by kaelabook**
# Description

Going off of Bibek's database, `/report` gives back the user's time in and time out for a specified time range, as well as the user's total hours worked and total pomodoros.

## Issues

Closes #390 

## Type of change

Select one or more of the following:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other (describe below)

# How Has This Been Tested?

- 1. run the /checkin report command
- 2a. (to see time in and time out) give the bot a time range formatted MM-DD-YYYY
- 2b. (to just see total hours and pomodoros) press enter
- 3. watch as the bot prints out a beautifully formatted report

This has not been tested for any edge cases, but needs to be fully tested on data that makes sense.

# Checklist:

- [x] All local commits have been pushed to remote
- [x] All changes on the base branch have been merged into this branch, either by rebase or merge
- [x] My code is [PEP-8](https://pep8.org/) compliant (excluding maximum line length, keep that to 100ish characters)
- [] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have added [Google Docstrings](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) for all new functions/methods
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Review by bibekrj — CHANGES_REQUESTED** [review]
Parts of the method needs to be functionalized as discussed previously.

File: Cogs/StudentCommands.py [hunk 1]
```diff
@@ -211,3 +212,50 @@ async def roll(self, interaction:discord.Interaction, roll:str):
...
+
+        with open(selected_song, 'r', encoding="utf-8") as file:
+            lyrics = file.read()
+
+        # Send song
+        await interaction.followup.send(f'Song #{number}:\n{lyrics}')
+
+        # Log command usage
+        await log(self.bot, f'{interaction.user} ran /sing_to_me in #{interaction.channel}')
+
```
Comment (BPennie03) [inline Cogs/StudentCommands.py:261]: This should probably be in a separate PR

**Review by BPennie03 — CHANGES_REQUESTED** [review]
😎

File: Cogs/StudentCommands.py (hunk 1 above)
Comment (kaelabook) [reply Cogs/StudentCommands.py:261]: no
